metrics/ci_integrations/docs/features/supported_database_version/01_supported_database_version.md

```diff
@@ -0,0 +1,88 @@
+
+According to the above, we should introduce a new `DatabaseVersionSyncStage` that is responsible for performing the required database compatibility checks, and run it at the beginning of the project metrics synchronization. If the CI Integrations tool does not support the current database version, or the database is updating, the synchronization must fail with the informative error message describing the required steps the user needs to perform to overcome the occurred issue. 
+
+To sum up, the synchronization algorithm now must include the `SyncStage`s in the following order:
+`DatabaseVersionSyncStage` -> `InProgressBuildsSyncStage` -> `NewBuildsSyncStage`
+
+## DatabaseVersionSyncStage
+The `DatabaseVersionSyncStage` needs the following values to be able to detect whether the CI Integrations tool is compatible with the current database:
+- [Supported database version](#Supported-Database-Version) of this CI Integrations tool;
+- [Database metadata](#Getting-Database-Metadata).
+
+Let's review the way of getting each of them separately:
+
+### Supported Database Version
+> Explain the process of getting the supported database version.
+
+Since the CI Integrations tool is built with the `SUPPORTED_DATABASE_VERSION` environment variable (based on the [Storing Database Metadata](https://github.com/platform-platform/monorepo/blob/master/metrics/docs/01_storing_database_metadata.md#supported-database-version) document), we can get this value in the application from the environment, using the `ApplicationMetadata` class from the [core](https://github.com/platform-platform/monorepo/tree/master/metrics/core) library (as described in this [section](https://github.com/platform-platform/monorepo/blob/master/metrics/web/docs/features/supported_database_version/01_supported_database_version.md#supported-database-version)).
+
+When we have the `SUPPORTED_DATABASE_VERSION` value, we need to provide it to the `DatabaseVersionSyncStage`.  To do that, we can add a `supportedDatabaseVersion` field to the `SyncConfig` class to store it and call the `DatabaseVersionSyncStage`s with the `SyncConfig` instance so the `SUPPORTED_DATABASE_VERSION` value is accessible.
+
+Let's proceed to the next [section](#Getting-Database-Metadata) and consider the ways of getting the Firestore database metadata in the CI Integrations tool.
+
+### Getting Database Metadata
+> Explain the way of loading the database metadata.
+
+The database metadata is represented by the `DatabaseMetadata` entity from the [`core` library](https://github.com/platform-platform/monorepo/tree/master/metrics/core). To be able to get the `DatabaseMetadata` we should introduce a new `fetchDatabaseMetadata` method to the `DestinationClient` and implement it in all `DestinationClient`'s subclasses. Since the `destinationClient` is available within all `SyncStage`'s subclasses, we can use it to fetch the database metadata within the `DatabaseVersionSyncStage` and perform any necessary compatibility checks.
```

Comment:
Does it make sense to create a separate interface for the `DestinationMetadata` in the `ci_integrations` and then create an adapter for the `DatabaseMeradata` to the `DestinationMetadata` interface?